server/lib/nl/fulfillment/fulfiller.py

```diff
@@ -233,3 +240,56 @@ def _prune_non_country_special_dc_vars(state: PopulateState):
                              list(dropped_vars))
 
   state.chart_vars_map = pruned_chart_vars_map
+
+
+#
+# Update chart vars map for toolformer rig mode because needs special handling
+# of topics where we only want the stat vars that match both the topic and the
+# actual query.
+#
+def _update_chart_vars_for_rig(state: PopulateState):
+  # set of all svs that are part of a topic
+  topic_svs = set()
+  for var, chart_vars_list in state.chart_vars_map.items():
+    if not var.startswith(_TOPIC_PREFIX):
+      continue
+    for cv in chart_vars_list:
+      topic_svs.update(cv.svs)
+
+  updated_chart_vars_map: Dict[str, List[ChartVars]] = {}
+
+  # remove topic chart vars
+  dropped_topics = set()
+  for var, chart_vars_list in state.chart_vars_map.items():
+    if var.startswith(_TOPIC_PREFIX):
+      dropped_topics.add(var)
+      continue
+    updated_chart_vars_map[var] = chart_vars_list
+
+  # add chart vars for detected svs with a score above model threshold & part
+  # of a topic that was in the original chart vars
+  added_svs = set()
+  for i, sv in enumerate(state.uttr.detection.svs_detected.single_sv.svs):
+    # skip topic svs
+    if sv.startswith(_TOPIC_PREFIX):
+      continue
+    # skip if score is below default threshold
+    if state.uttr.detection.svs_detected.single_sv.scores[
+        i] < state.uttr.detection.svs_detected.model_threshold:
+      continue
+    # skip if not part of a topic that was detected
+    if not sv in topic_svs:
+      continue
+    updated_chart_vars_map[sv] = updated_chart_vars_map.get(sv, [])
+    updated_chart_vars_map[sv].append(
+        ChartVars(svs=[sv], orig_sv_map={sv: [sv]}))
```

Comment:
Hmmm, we don't want to add it if it is already there?

Concretely, should we avoid re-adding SVs that are already there in chart_vars.svs?

Reply:
as discussed offline, don't add if sv is already a key in chart vars